tests/test_io.py

```diff
@@ -13,4 +14,22 @@ def test_write_image(tmpdir):
     print(tmpdir/"p.png")
     assert os.path.isfile(tmpdir/"p.png")
 
+def test_write_vtk(tmpdir):
+    #pytest.skip("matplotlib not working")
+    #dtype, device = dtype_device
+    tmpdirr = "/Users/mariobedrunka/Documents/10_science/10_lattice_boltzmann/10_simulation/10_lettuce/data"
+    lattice = Lattice(D2Q9, "cpu")
+    flow = TaylorGreenVortex2D(resolution=16, reynolds_number=10, mach_number=0.05, lattice=lattice)
+    p, u = flow.initial_solution(flow.grid)
+    point_dict = {}
+    point_dict["p"] = p[0, ..., None]
+    write_vtk(point_dict, 1, filename_base=tmpdir, filename="/output")
+    #print(tmpdir/"output_1.vtr")
+    assert os.path.isfile(tmpdir/"output_1.vtr")
```

Comment:
What is the difference between vtr and vtk? Which one should be used here?

Reply:
.vtr is more like a file type of VTK. The type depends on the 'grid class' you're using:

VtkImageData        = VtkFileType("ImageData", ".vti")
VtkPolyData         = VtkFileType("PolyData", ".vtp")
VtkRectilinearGrid  = VtkFileType("RectilinearGrid", ".vtr")
VtkStructuredGrid   = VtkFileType("StructuredGrid", ".vts")
VtkUnstructuredGrid = VtkFileType("UnstructuredGrid", ".vtu")